platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapService.java

```diff
@@ -54,10 +53,8 @@ public LdapService(final BundleContext context) {
      * @param newModule that will replace a module or be added to the list of modules.
      */
     public synchronized void update(Module newModule) {
-        modules = modules.stream()
-                .filter(m -> !m.getName()
-                        .equals(newModule.getName()))
-                .collect(Collectors.toList());
+        modules.removeIf(m -> m.getName()
```

Comment:
1. I'm not generally a fan of mutable collections, though the fact that this method is synchronized (and, I assume, all the other methods that modify the `modules` list are as well) helps.
2. However, you're risking a future problem if someone changes the implementation class of `modules` to a Collection whose iterator doesn't support `remove`. It might be best to change its declaration to `ArrayList` to be specific. And, while you're there, you could also make it `final` since you're no longer replacing it but mutating it.